src/filetypes/checktype.c

```diff
@@ -13,18 +13,12 @@ enum check_type_return
     Unknown
 };
 
-int laser_checktype(const char *filename, const struct laser_filetype formats[])
+int laser_checktype_sniff(int fd, const char *filename,
```

Comment:
I'd feel that it's better to have `laser_checktype` function that first checks on the extension, if nothing is matched then we go for the file signatures.